### ISSUE TITLE
Count overlapping circles  

### ISSUE THREAD

**Issue by elboyran** [NLeSC/XAI#44]

**Comment by elboyran** [comment]
e-mail from 03.04.2019

> Hi all,
> 
> I updated the code so that now also Eric's data about counting circles can be trained (and the other datasets) by uncommenting the right lines in setting.py and adding the CountingCircles data in the datafolder XAI\Data\CountingCircles. 
> 
> The training of the neural network got stuck around 42% accuracy on the validation data. Basically, its strategy is to classify each image as "1 circle" (which is around 42% of the validation set the case). It seems that these floating, varying in numbers, circles are too difficult to classify with a simple multi-layered neural network setup and more advanced CNN-techniques are required. A research question may be to look for the point where the data becomes too difficult: rotation_and_scaled_triangles_and_squares = doable by a neural net, whereas the counting_circles = not doable by a neural net. Where in between does it become too difficult for the net and what makes this so difficult? Can this be mathematically justified?
> 
> I wish you all a fruitful meeting tomorrow and please keep me posted :-).
> 
> Kind regards,
> 
> Joost

**Comment by elboyran** [comment]
Related commits: 
12d15a5bbfd71dfaeeac35fc1ccb888c644daee4
0b0a7c6ad50e44939ff092dc4041d6c320b2998b
